sotodlib/site_pipeline/update_smurf_caldbs.py

```diff
@@ -109,15 +131,309 @@ def smurf_detset_info(config:str | dict,
+            continue
+
+        ridx = ridx[0]
+        cal.bg = iva['bgmap'][ridx]
+        cal.polarity = iva['polarity'][ridx]
+        cal.r_n = iva['R_n'][ridx]
+        cal.p_sat = iva['p_sat'][ridx]
+
+    obs_biases = dict(zip(am.bias_lines.vals, am.biases[:, 0] * 2*rtm_bit_to_volt))
+    bias_line_is_valid = {k: True for k in obs_biases.keys()}
+
+    # check to see if biases have changed between bias steps and obs
+    for bsa in bsas.values():
+        if bsa is None:
+            continue
+
+        for bg, vb_bsa in enumerate(bsa['Vbias']):
+            bl_label = f"{bsa['meta']['stream_id']}_b{bg:0>2}"
+            if np.isnan(vb_bsa):
+                bias_line_is_valid[bl_label] = False
+                continue
+
+            if np.abs(vb_bsa - obs_biases[bl_label]) > 0.1:
+                logger.debug("bias step and obs biases don't match for %s", bl_label)
+                bias_line_is_valid[bl_label] = False
+
+    for i, cal in enumerate(cals):
+        band = am.det_info.smurf.band[i]
+        chan = am.det_info.smurf.channel[i]
+        detset = am.det_info.detset[i]
+        stream_id = am.det_info.stream_id[i]
+        bg = cal.bg
+        bsa = bsas[detset]
+
+        if bsa is None or bg == -1:
+            continue
+
+        bl_label = f'{stream_id}_b{bg:0>2}'
+        if not bias_line_is_valid[bl_label]:
+            continue
+
+        ridx = np.where(
+            (bsa['bands'] == band) & (bsa['channels'] == chan)
+        )[0]
+        if not ridx: # Channel doesn't exist in bias step analysis
+            continue
+
+        ridx = ridx[0]
+        cal.r_tes = bsa['R0'][ridx]
+        cal.r_frac = bsa['Rfrac'][ridx]
+        cal.p_bias = bsa['Pj'][ridx]
+        cal.s_i = bsa['Si'][ridx]
+        cal.phase_to_pW = pA_per_phi0 / (2*np.pi) / cal.s_i * cal.polarity
+
+    rset = core.metadata.ResultSet.from_friend(np.array(
+        [astuple(c) for c in cals], dtype=cal_dtype
+    ))
+    return rset
+
+
+def get_obs_with_detsets(ctx, detset_idx):
+    """Gets all observations with type 'obs' that have detset data"""
+    db = core.metadata.ManifestDb(detset_idx)
+    detsets = db.get_entries(['dataset'])['dataset']
+    obs_ids = set()
+    for dset in detsets:
+        obs_ids = obs_ids.union(ctx.obsfiledb.get_obs_with_detset(dset))
+    return obs_ids
+
+
+def update_det_caldb(ctx, idx_path, detset_idx, h5_path, logger=None, 
+                     show_pb=False, format_exc=False, overwrite=True):
+    if logger is None:
+        logger = default_logger
+
+    if not os.path.exists(idx_path):
+        scheme = core.metadata.ManifestScheme()
+        scheme.add_exact_match('obs:obs_id')
+        scheme.add_data_field('dataset')
+        db = core.metadata.ManifestDb(scheme=scheme)
+        db.to_file(idx_path)
+    db = core.metadata.ManifestDb(idx_path)
+    
+    # detset_db = metadata.Manifest(detset_idx)
+    existing_obsids = set(db.get_entries(['dataset'])['dataset'])
```

Comment:
don't bother to check for existing entries if overwrite=True?

Reply:
If overwrite=True, do we want to re-run the calibration analysis for all obs_ids? I thought that we just wanted it to not care if the h5 entry already existed...

Reply:
I think we're thinking of overwrite doing different things. I really do mean re-run the calibration analysis 

There have been times when I've done this for other manifests, that's why I like to have the option around. But I realize that has usually be for analyzes where it's usually only running on sections of the database so there are times when I have to go say "rerun on all of last weeks data" but I wouldn't want to just delete the files and start again. 

I don't think just overwriting the .h5 entry is something that should ever happen in the current setup.

Reply:
ah I see.... i think most recent commit fixes it so that overwrite means the same thing as the detset function